src/Microsoft.AspNetCore.HttpOverrides/IPNetwork.cs

```diff
@@ -1,12 +1,245 @@
+
+        /// <summary>
+        /// All IPv4 addresses 0.0.0.0/0
+        /// </summary>
+        public static readonly IPNetwork AllIPv4 = Parse("0.0.0.0/0");
+
+        /// <summary>
+        /// IPv4 localhost range 127.0.0.0/8 (255.0.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Loopback = Parse("127.0.0.1/8");
+
+        /// <summary>
+        /// IPv4 link-local range 169.254.0.0/16 (255.255.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4LinkLocal = Parse("169.254.0.0/16");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 10.0.0.0/8 (255.0.0.0)
+        /// </summary>
+        /// <seealso cref="RFC1918" />
+        public static readonly IPNetwork IPv4Private10 = Parse("10.0.0.0/8");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 172.16.0.0/12 (255.240.0.0)
+        /// </summary>
+        /// <seealso cref="RFC1918" />
+        public static readonly IPNetwork IPv4Private172 = Parse("172.16.0.0/12");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 192.168.0.0/16 (255.255.0.0)
+        /// </summary>
+        /// <seealso cref="RFC1918" />
+        public static readonly IPNetwork IPv4Private192 = Parse("192.168.0.0/16");
+
+        /// <summary>
+        /// IPv4 Multicast range 224.0.0.0/4
+        /// </summary>
+        public static readonly IPNetwork IPv4Multicast = Parse("224.0.0.0/4");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 Addresses 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16
+        /// </summary>
+        /// <seealso cref="IPv4PrivateNetworks" />
+        public static readonly ImmutableArray<IPNetwork> RFC1918 = ImmutableArray.Create(IPv4Private10, IPv4Private172, IPv4Private192);
+
+        /// <summary>
+        /// <see cref="IPv4LinkLocal" />
+        /// <see cref="IPv4Loopback" />
+        /// <see cref="RFC1918" />
+        /// </summary>
+        /// <seealso cref="PrivateNetworks" />
+        public static readonly ImmutableArray<IPNetwork> IPv4PrivateNetworks = ImmutableArray.Create(IPv4Loopback, IPv4LinkLocal, IPv4Private10, IPv4Private172, IPv4Private192);
+
+        /// <summary>
+        /// IPv6 default unicast route ::/0 (everything)
+        /// </summary>
+        public static readonly IPNetwork AllIPv6 = Parse("::/0");
+
+        /// <summary>
+        /// IPv6 localhost range ::1/128
+        /// </summary>
+        public static readonly IPNetwork IPv6Loopback = Parse("::1/128");
+
+        /// <summary>
+        /// Private IPv6 range fc00::/7 (RFC1918 equivalent)
+        /// </summary>
+        /// <seealso cref="PrivateNetworks" />
+        /// <seealso cref="IPv4PrivateNetworks" />
+        /// <seealso cref="IPv6PrivateNetworks" />
+        public static readonly IPNetwork IPv6Private = Parse("fc00::/7");
+
+        /// <summary>
+        /// IPv6 Teredo range 2001::/32
+        /// </summary>
+        public static readonly IPNetwork Teredo = Parse("2001::/32");
```

Comment:
Let's restrict the constants to the ones that will be commonly used in the forwarded headers scenario. I would not expect Teredo, Multicast, etc. to be used in these scenarios.